skimage/io/tests/test_imageio.py

```diff
@@ -65,7 +66,12 @@ def test_imsave_roundtrip(self, shape, dtype, tmp_path):
             min_, max_, endpoint=True, num=np.prod(shape), dtype=dtype
         )
         expected = expected.reshape(shape)
-        file_path = tmp_path / "roundtrip.png"
+        if hasattr(threading, 'get_native_id'):
+            tmp_dir = tmp_path / str(threading.get_native_id())
+        else:
+            tmp_dir = tmp_path / str(threading.get_ident())
+        tmp_dir.mkdir()
+        file_path = tmp_dir / "roundtrip.png"
```

Comment:
I assume `tmp_path` isn't thread-safe. Could we conditionally override it in `conftest.py` with a thread-safe version? That way we would avoid future bugs because I would certainly forget that I needed to do this in one year.

Reply:
Ensuring a different `tmp_path` per thread is not possible due to a limitation in `pytest-run-parallel`: https://github.com/Quansight-Labs/pytest-run-parallel/issues/14